script/templates/CreateMorphoMarket.s.sol

```diff
@@ -0,0 +1,314 @@
+import {AggregatorV3Interface} from "@protocol/oracles/AggregatorV3Interface.sol";
+import {IERC4626} from "@forge-std/interfaces/IERC4626.sol";
+
+/// @notice Template script: Create a Morpho Blue market and configure a vault cap/supply queue based on JSON
+contract CreateMorphoMarket is Script, Test {
+    using ChainIds for uint256;
+    using stdJson for string;
+
+    struct MarketConfig {
+        string vaultAddressName; // vault registry name to configure caps/queues
+        string loanTokenName; // name key for loan token (Addresses)
+        string collateralTokenName; // name key for collateral token (Addresses)
+        string irmName; // name key for IRM (Addresses)
+        uint256 lltv; // loan-to-value mantissa
+        uint256 supplyCap; // cap in loan token units
+        bool setSupplyQueue; // if true, update supply queue to only this market
+        bool autoDeposit; // if true, perform deposit/supply/borrow demo actions
+        uint256 vaultDepositAssets; // amount of assets to deposit into the vault
+        uint256 collateralAmount; // amount of collateral to supply to Morpho
+        uint256 borrowAssets; // amount of assets to borrow
+        // Optional oracle deployment block: presence triggers deployment if missing
+        bool oracleConfigPresent;
+        OracleConfig oracle;
+    }
+
+    struct OracleConfig {
+        string addressName; // name key for oracle (Addresses)
+        string baseFeedName; // e.g. CHAINLINK_WELL_USD
+        uint8 baseFeedDecimals; // e.g. 18
+        string quoteFeedName; // e.g. CHAINLINK_USDC_USD
+        uint8 quoteFeedDecimals; // e.g. 6
+    }
+
+    uint256 internal constant MARKET_PARAMS_BYTES_LENGTH = 5 * 32;
+
+    function run() external {
+        // Setup fork for Base chain
+        BASE_FORK_ID.createForksAndSelect();
+
+        string memory json = vm.readFile(vm.envString("NEW_MARKET_PATH"));
+        MarketConfig memory cfg = _parseConfig(json);
+
+        Addresses addresses = new Addresses();
+
+        vm.startBroadcast();
+
+        // Ensure oracle is deployed or deploy it
+        _ensureOracle(addresses, cfg.oracleConfigPresent, cfg.oracle);
+
+        MarketParams memory market = _buildMarketParams(addresses, cfg);
+        bytes32 marketId = _computeMarketId(market);
+        address morphoBlue = addresses.getAddress("MORPHO_BLUE");
+
+        _createMarket(morphoBlue, market, marketId, cfg);
+
+        // TODO: test the below after a valid market creation
+
+        if (bytes(cfg.vaultAddressName).length != 0 && cfg.supplyCap > 0) {
+            _configureVault(addresses, cfg.vaultAddressName, cfg.supplyCap, cfg.setSupplyQueue, marketId, market);
+        }
+
+        if (cfg.autoDeposit && bytes(cfg.vaultAddressName).length != 0) {
+            _demoFlow(addresses, cfg.vaultAddressName, morphoBlue, market, cfg.vaultDepositAssets, cfg.collateralAmount, cfg.borrowAssets);
```

Comment:
I don't think this should be optional. This should be enforced , check https://docs.morpho.org/curate/tutorials-market-v1/creating-market/#fill-all-attributes

<img width="772" height="415" alt="Image" src="https://github.com/user-attachments/assets/51e49ea1-14a1-4460-9cfc-2bdb12aca165" />

The problem is we need Anthias to activate the market in the vault now because we already deployed the vault and gave them the curator role. Let me sync with them